cf-java-logging-support-core/src/main/java/com/sap/hcp/cf/logging/common/LogContext.java

```diff
@@ -112,11 +112,14 @@ private static void setOrGenerateCorrelationId(String correlationId) {
         }
     }
 
+    private static class LoggerHolder {
```

Comment:
I totally understand the holder pattern. But why not just use a static field instead? This is what is done in most other cases not addressed by this PR.

Reply:
I'll change it. Rationale in my second comment.

Reply:
We found out the hard way, that the LogContext will break the test with a static field. The holder pattern will be used to mitigate this.